From bcb288c5ef29e38095830460017b54774a679513 Mon Sep 17 00:00:00 2001
From: Will Thompson <will@willthompson.co.uk>
Date: Sat, 15 Aug 2020 07:39:06 +0100
Subject: [PATCH 3/3] Install desktop file and icons

xdg-icon-resource and xdg-desktop-menu are not available in the SDK.
---
 Makefile | 23 +++++------------------
 1 file changed, 5 insertions(+), 18 deletions(-)

diff --git a/Makefile b/Makefile
index 7fe1cf1e..cadea5a9 100644
--- a/Makefile
+++ b/Makefile
@@ -826,24 +826,11 @@ install-nokia770:
 install-xdg:
 	@echo
 	@echo "...Installing launcher icon into desktop environment..."
-	@if [ "x$(shell which xdg-icon-resource install)" != "x" ]; then \
-	  xdg-icon-resource install --size 192 data/images/icon192x192.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 128 data/images/icon128x128.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 96 data/images/icon96x96.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 64 data/images/icon64x64.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 48 data/images/icon48x48.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 32 data/images/icon32x32.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 22 data/images/icon22x22.png tux4kids-tuxpaint ; \
-	  xdg-icon-resource install --size 16 data/images/icon16x16.png tux4kids-tuxpaint ; \
-	fi
-	@if [ "x$(shell which xdg-desktop-menu)" != "x" ]; then \
-	  cp src/tuxpaint.desktop ./tux4kids-tuxpaint.desktop ; \
-	  xdg-desktop-menu install tux4kids-tuxpaint.desktop ; \
-	  rm ./tux4kids-tuxpaint.desktop ; \
-	fi
-	@if [ "x$(shell which update-desktop-database)" != "x" ]; then \
-	  update-desktop-database ; \
-	fi
+	install -m644 -D data/images/tuxpaint-icon.svg $(DESTDIR)$(PREFIX)/share/icons/hicolor/scalable/apps/tuxpaint.svg
+	for size in 192 128 96 64 48 32 22 16; do \
+	  install -m644 -D data/images/icon$${size}x$${size}.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/$${size}x$${size}/apps/tuxpaint.png; \
+	done
+	install -m644 -D src/tuxpaint.desktop $(DESTDIR)$(PREFIX)/share/applications/tuxpaint.desktop
 
 # Install the PNG icon (for KDE desktop, etc.)
 # and the 24-color 32x32 XPM (for other Window managers):
-- 
2.26.2

